src/app/signers/data/UseSignerAddresses.ts

```diff
@@ -27,15 +37,59 @@ export function useGetStackersBySignerQuery() {
   });
 }
 
-export function useSuspenseSignerAddresses(cycleId: number, signerKey: string) {
+export function useSuspenseSignerStackers(cycleId: number, signerKey: string) {
   const { url: activeNetworkUrl } = useGlobalContext().activeNetwork;
 
-  return useSuspenseQuery<SignersStackersData>({
-    queryKey: [SIGNER_ADDRESSES_QUERY_KEY, cycleId, signerKey],
+  return useSuspenseQuery<ApiResponseWithResultsOffset<SignersStackersData>>({
+    queryKey: [SIGNER_STACKERS_QUERY_KEY, cycleId, signerKey],
     queryFn: () =>
-      fetch(`${activeNetworkUrl}/extended/v2/pox/cycles/${cycleId}/signers/${signerKey}`).then(
-        res => res.json()
-      ),
+      fetch(
+        `${activeNetworkUrl}/extended/v2/pox/cycles/${cycleId}/signers/${signerKey}/stackers`
+      ).then(res => res.json()),
     staleTime: TEN_MINUTES,
   });
 }
+
+const DEFAULT_LIST_LIMIT = 10;
+
+const fetchStackers = async (
+  apiUrl: string,
+  cycleId: number,
+  signerKey: string,
+  pageParam: number,
+  options: any
+): Promise<GenericResponseType<SignersStackersData>> => {
+  const limit = options.limit || DEFAULT_LIST_LIMIT;
+  const offset = pageParam || 0;
+  const queryString = new URLSearchParams({
+    limit: limit.toString(),
+    offset: offset.toString(),
+  }).toString();
+  const response = await fetch(
+    `${apiUrl}/extended/v2/pox/cycles/${cycleId}/signers/${signerKey}/stackers${
+      queryString ? `?${queryString}` : ''
+    }`
+  );
+  return response.json();
+};
+
+export function useSuspenseSignerStackersInfinite(
+  cycleId: number,
+  signerKey: string,
+  options: any = {}
+): UseInfiniteQueryResult<InfiniteData<GenericResponseType<SignersStackersData>>> {
+  // const api = useApi();
+  // if (!address) throw new Error('Address is required');
+  const { url: activeNetworkUrl } = useGlobalContext().activeNetwork;
+
+  return useInfiniteQuery<GenericResponseType<SignersStackersData>>({
+    queryKey: [SIGNER_STACKERS_INFINITE_QUERY_KEY, cycleId, signerKey],
```

Comment:
should options be added to query key so e.g. different limit would trigger a different request?

Reply:
Pretty sure react query handles caching the paginated data internally through its pages and pageParams structure. Looking at the infinite query example from the docs:
The pagination parameters (pageParam, offset, limit) don't need to be in the queryKey because:
The pageParam is managed internally by the infinite query
Each "page" of data is stored in data.pages
The pagination state is tracked in data.pageParams
I really dont think it would be likely where we would have this exact same query key but with just a different limit